### PR TITLE
Elva homepage

### PR DESCRIPTION
**Pull request by XiaoyananWang**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ import LandingPage from "./components/pages/landing-page/LandingPage";
 import LoginPage from "./components/pages/authentication-pages/LoginPage";
 import PasswordRecoverPage from "./components/pages/authentication-pages/PasswordRecoveryPage";
 import SignupPage from "./components/pages/authentication-pages/SignupPage";
+import Profile from "./components/pages/profile-page/ProfilePage";
+import CompletedPage from "./components/pages/completed-page/CompletedPage";
+import SettingsPage from "./components/pages/settings-page/SettingsPage";
```
Comment (WhitneyLiu) [inline client/src/App.js:8]: 注意按首字母排序

File: client/src/components/pages/authentication-pages/PasswordRecoveryPage.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ export default function PasswordRecoverPage() {
               type="password"
               label="Confirm password:"
             />
-            <SubmitButton text="Send verification code" />
+            <SubmitButton text="Set password" />
```
Comment (WhitneyLiu) [inline client/src/components/pages/authentication-pages/PasswordRecoveryPage.js:131]: 我觉得Send verification code比较user friendly

File: client/src/components/pages/home-page/components/HomepageHeader.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import Button from '../../../sharable-components/Button';
+import Container from '../../../sharable-components/Container';
+import Logo from '../../../sharable-components/Logo';
+
+export default function HomepageHeader({ onLogout }) {
```
Comment (WhitneyLiu) [inline client/src/components/pages/home-page/components/HomepageHeader.js:5]: 我觉得更好的排版方式是把logo放在sidenavbar最上面，logout放在最下面的user action里

File: client/src/components/pages/home-page/components/Sidebar.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { useSelector } from "react-redux";
+import { Link } from "react-router-dom";
+import Container from "../../../sharable-components/Container";
+import {
```
Comment (WhitneyLiu) [inline client/src/components/pages/home-page/components/Sidebar.js:4]: import注意排序

File: client/src/components/pages/home-page/components/Sidebar.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  { name: "Profile", href: "/profile", icon: UsersIcon, current: false },
+  { name: "Completed", href: "/completed", icon: FolderIcon, current: false },
+  { name: "Settings", href: "/settings", icon: ChartPieIcon, current: false },
+];
+
+function classNames(...classes) {
+  return classes.filter(Boolean).join(" ");
+}
+
+export default function Sidebar({ profileImage }) {
```
Comment (WhitneyLiu) [inline client/src/components/pages/home-page/components/Sidebar.js:23]: 我看了一下production的网页，我在想是不是用tailwind紫色的那个sidnav更好看一些？还有就是为了防我们的网页颜色theme保持一致，最好把他们原始的紫色改成蓝色，具体怎么改，直接search “indigo” 全都改成 “blue” 就可以了。我检查了一下，发现sidenav current page的focus不太对，可能需要检查一下

File: client/src/components/sharable-components/Layout.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import { useDispatch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { logout } from "../../redux/reducer/authenticationReducer";
+import HomepageHeader from "../pages/home-page/components/HomepageHeader";
+import Greeting from "../pages/home-page/components/Greeting";
+import { useSelector } from "react-redux";
+import Sidebar from "../pages/home-page/components/Sidebar";
+import Container from "./Container";
+
+export default function Layout({ children }) {
```
Comment (WhitneyLiu) [inline client/src/components/sharable-components/Layout.js:11]: 我看了网页效果，可能素了点，可以参考Feature.js里面的backgroundImage给body的部分加一个background

File: client/src/redux/store.js [hunk 7]
```diff
@@ -1,10 +1,12 @@
 import authenticationReducer from "./reducer/authenticationReducer";
 import notificationReducer from "./reducer/notificationReducer";
 import { configureStore } from "@reduxjs/toolkit";
```
Comment (WhitneyLiu) [inline client/src/redux/store.js:3]: 注意排序

File: client/src/App.js (hunk 1 above)
Comment (XiaoyananWang) [reply client/src/App.js:8]: 好的！

File: client/src/components/pages/authentication-pages/PasswordRecoveryPage.js (hunk 2 above)
Comment (XiaoyananWang) [reply client/src/components/pages/authentication-pages/PasswordRecoveryPage.js:131]: 这里我昨天试的时候发现第二个按钮还是send verification code，但是这个步骤是在send verification code之后了，到了修改密码的那一步，底下的按钮还是send verification code，昨天试的时候突然有点迷惑。 或许改成set password比较好，就是在send verification code之后的那个步骤。

File: client/src/components/pages/home-page/components/HomepageHeader.js (hunk 3 above)
Comment (XiaoyananWang) [reply client/src/components/pages/home-page/components/HomepageHeader.js:5]: 好的，我修改为这个方式

File: client/src/components/pages/home-page/components/Sidebar.js (hunk 4 above)
Comment (XiaoyananWang) [reply client/src/components/pages/home-page/components/Sidebar.js:4]: 好滴

File: client/src/components/sharable-components/Layout.js (hunk 6 above)
Comment (XiaoyananWang) [reply client/src/components/sharable-components/Layout.js:11]: 好的，我之后修改的时候加上

File: client/src/components/pages/authentication-pages/PasswordRecoveryPage.js (hunk 2 above)
Comment (WhitneyLiu) [reply client/src/components/pages/authentication-pages/PasswordRecoveryPage.js:131]: 你说的对，怪我没看仔细

File: client/src/components/pages/home-page/components/Sidebar.js (hunk 5 above)
Comment (WhitneyLiu) [reply client/src/components/pages/home-page/components/Sidebar.js:23]: 紫色模版的码在这里。import {
  CalendarIcon,
  ChartPieIcon,
  DocumentDuplicateIcon,
  FolderIcon,
  HomeIcon,
  UsersIcon,
} from '@heroicons/react/24/outline'

const navigation = [
  { name: 'Dashboard', href: '#', icon: HomeIcon, count: '5', current: true },
  { name: 'Team', href: '#', icon: UsersIcon, current: false },
  { name: 'Projects', href: '#', icon: FolderIcon, count: '12', current: false },
  { name: 'Calendar', href: '#', icon: CalendarIcon, count: '20+', current: false },
  { name: 'Documents', href: '#', icon: DocumentDuplicateIcon, current: false },
  { name: 'Reports', href: '#', icon: ChartPieIcon, current: false },
]
const teams = [
  { id: 1, name: 'Heroicons', href: '#', initial: 'H', current: false },
  { id: 2, name: 'Tailwind Labs', href: '#', initial: 'T', current: false },
  { id: 3, name: 'Workcation', href: '#', initial: 'W', current: false },
]

function classNames(...classes) {
  return classes.filter(Boolean).join(' ')
}

export default function Example() {
  return (
    <div className="flex grow flex-col gap-y-5 overflow-y-auto bg-indigo-600 px-6">
      <div className="flex h-16 shrink-0 items-center">
        Your Company
      </div>
      <nav className="flex flex-1 flex-col">
        <ul role="list" className="flex flex-1 flex-col gap-y-7">
          <li>
            <ul role="list" className="-mx-2 space-y-1">
              {navigation.map((item) => (
                <li key={item.name}>
                  <a
                    href={item.href}
                    className={classNames(
                      item.current
                        ? 'bg-indigo-700 text-white'
                        : 'text-indigo-200 hover:text-white hover:bg-indigo-700',
                      'group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold'
                    )}
                  >
                    <item.icon
                      className={classNames(
                        item.current ? 'text-white' : 'text-indigo-200 group-hover:text-white',
                        'h-6 w-6 shrink-0'
                      )}
                      aria-hidden="true"
                    />
                    {item.name}
                    {item.count ? (
                      <span
                        className="ml-auto w-9 min-w-max whitespace-nowrap rounded-full bg-indigo-600 px-2.5 py-0.5 text-center text-xs font-medium leading-5 text-white ring-1 ring-inset ring-indigo-500"
                        aria-hidden="true"
                      >
                        {item.count}
                      </span>
                    ) : null}
                  </a>
                </li>
              ))}
            </ul>
          </li>
          <li>
            <div className="text-xs font-semibold leading-6 text-indigo-200">Your teams</div>
            <ul role="list" className="-mx-2 mt-2 space-y-1">
              {teams.map((team) => (
                <li key={team.name}>
                  <a
                    href={team.href}
                    className={classNames(
                      team.current
                        ? 'bg-indigo-700 text-white'
                        : 'text-indigo-200 hover:text-white hover:bg-indigo-700',
                      'group flex gap-x-3 rounded-md p-2 text-sm leading-6 font-semibold'
                    )}
                  >
                    <span className="flex h-6 w-6 shrink-0 items-center justify-center rounded-lg border border-indigo-400 bg-indigo-500 text-[0.625rem] font-medium text-white">
                      {team.initial}
                    </span>
                    <span className="truncate">{team.name}</span>
                  </a>
                </li>
              ))}
            </ul>
          </li>
          <li className="-mx-6 mt-auto">
            <a
              href="#"
              className="flex items-center gap-x-4 px-6 py-3 text-sm font-semibold leading-6 text-white hover:bg-indigo-700"
            >
              
              <span className="sr-only">Your profile</span>
              <span aria-hidden="true">Tom Cook</span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  )
}

Comment (WhitneyLiu) [reply client/src/components/pages/home-page/components/Sidebar.js:23]: 对滴

**Pull request merged**